### ISSUE TITLE
delete Cache not working

### ISSUE THREAD

**Issue by 810** [getgrav/grav-plugin-admin#248]
Not working:
"All cache"
"Cache only"

Working:
"Assets only"
"Images only"

error message:

invalid ajax respons.

Browsers:
IE 11 tested
Edge tested

**Comment by rhukster** [comment]
As mentioned before. The admin has only been tested in Safari, Chrome, Firefox.  IE is not at the top of our list at the moment as we're mac-centric.  We'll get to it though.  For now please use one of these other browsers.

**Comment by w00fz** [comment]
Would be also useful if you could inspect the network response in IE and report back the full body of it.

**Comment by 810** [comment]
its not only Ie, also chrome, opera, ff.

**Comment by w00fz** [comment]
Works fine for me.

![dashboard grav 2015-11-20 12-55-59](https://cloud.githubusercontent.com/assets/11734/11311664/1a7d064c-8f86-11e5-8793-b5651ae632e4.png)

This was 'All Cache'. Could you attach the network response as explained in my previous post?
Also what is your setup? Localserver? Apache / IIS (you are on windows I assume)?

**Comment by 810** [comment]
HTTP/1.1 304 Not Modified
Date: Fri, 20 Nov 2015 21:00:52 GMT
Connection: keep-alive
Pragma: no-cache
Cache-Control: max-age=604800
Expires: Fri, 27 Nov 2015 21:00:52 GMT
Etag: 17bee57b5149aabdf274a21e3b020268
Vary: Accept-Encoding
Set-Cookie: (removecookieid) expires=Fri, 20-Nov-2015 21:30:52 GMT; Max-Age=1800; path=/docs2
Last-Modified: Sun, 01 Nov 2015 02:26:30 GMT
Host-Header: 192fc2e7e50945beb8231a492d6a8024
X-Proxy-Cache: MISS
Server: cloudflare-nginx
cf-ray: 2487163de9342b4c-AMS

GET /docs2/admin/cache.json/task:clearCache/cleartype:all HTTP/1.1
Host: jellekoksolutions.nl
User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:44.0) Gecko/20100101 Firefox/44.0
Accept: application/json, text/javascript, */*; q=0.01
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Referer: http://jellekoksolutions.nl/docs2/admin
X-Requested-With: XMLHttpRequest
Cookie: grav-site-removecookieid; grav-site-49ee3f1=gsc14kpl88go7ob5ej24su2dr1; __cfduid=d4f6be8c04d843281797d32470473fb7c1443754988
Connection: keep-alive
If-Modified-Since: Sun, 01 Nov 2015 02:26:30 GMT
If-None-Match: 17bee57b5149aabdf274a21e3b020268

**Comment by w00fz** [comment]
Those are the headers, could you paste the actual body response?

**Comment by hugoaf** [comment]
I have exact same problem when selecting chache only or all cache methods, this issue only occurs in public server at siteground, (does not occur in my windows local enviroment). Cleaning cache in cli works ok.

In my case ajax response is empty:
![invalid-ajax-response](https://cloud.githubusercontent.com/assets/8611472/11326973/fee482f4-912c-11e5-9bda-cd9e474e7ab7.jpg)

As a side note, if right after getting this error message, if I go to Themes, I get a To many open files error.

![to_many_open_files](https://cloud.githubusercontent.com/assets/8611472/11326982/46d80428-912d-11e5-920e-3bd62b45d265.jpg)

Could be a server limitation?

**Comment by w00fz** [comment]
Looks like it is, if you have ssh access could you run the command `ulimit -n` ?
This is mine on OSX:

```
🌀> ulimit -n
4864
```

I think it's something we should catch, requires further investigation.

**Comment by hugoaf** [comment]
well, you have a lot more than I do in sitegrownd growbig shared account
```
# ulimit -n
1024
```
BTY I added the line to convert the `RecursiveIteratorIterator` to array with `$files = iterator_to_array($files);` but error still.

**Comment by hugoaf** [comment]
Ok, these are the test I have done:
1. copied whole site to another server, problem does not occur in hostgator
2. did a clean install in same server and copied user folder, and problem did not occur
3. went back to original installation (the one with the problem) to check folders and files permissions in cache, all have 755 and 644 respectively, with same owner and group as the rest of the grav site.
4. performed a clear-cache from cli
5. and finally problem was solved after remove by hand the remaining files and folders inside cache dir. Doing a `clear-cache --all` from cli also solves the issue.

Problem solved to me, cache is working and clearing cache from admin works as expected.

With further tests I was able to replicate the issue, uploading the cache folder from my local (windows wamp) to the linux server, the clear cache button fails on clear all, the `clear-cache` command from cli leaves some files and folders, but `clear-cache --all` removes all of them.

Will be great to know if @810 has solved the issue too.
